src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHubSearcher.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NuGet.Jobs.GitHubIndexer
+{
+    public class GitHubSearcher : IGitRepoSearcher
+    {
+        public const int MIN_STARS = 100;
+        public const int RESULTS_PER_PAGE = 100; // Maximum is 100 :(
+        private IGitHubClient _client;
+
+        public GitHubSearcher(IGitHubClient client)
+        {
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+        }
+
+        private async Task<List<Repository>> GetResultsForPage(int currPage, int totalCount, int maxStarCount = -1, string lastRecordName = null)
+        {
+            if (_client.GetLastApiInfo() != null && _client.GetLastApiInfo().RateLimit.Remaining == 0)
+            {
+                Console.WriteLine("Waiting a minute to cooldown..");
+                await Task.Delay(TimeSpan.FromSeconds(61));
+                Console.WriteLine("Resuming search.");
+            }
+
+            var request = new SearchRepositoriesRequest
+            {
+                Stars = maxStarCount == -1 ? Range.GreaterThan(MIN_STARS) : new Range(MIN_STARS, maxStarCount),
+                Language = Language.CSharp,
+                SortField = RepoSearchSort.Stars,
+                Order = SortDirection.Descending,
+                PerPage = RESULTS_PER_PAGE,
+                Page = currPage
+            };
+
+            List<Repository> resultList = new List<Repository>();
```

Comment:
Please use `var` everywhere:

```suggestion
            var resultList = new List<Repository>();
```